src/server/rpc/storage.rs

```diff
@@ -2387,13 +2477,25 @@ mod tests {
                 .into_iter()
                 .map(|r| r.values.into_iter())
                 .flatten()
-                .map(|v| String::from_utf8(v).expect("string value response was not utf8"))
+                .map(Self::bytes_to_strings)
                 .collect::<Vec<_>>();
 
             strings.sort();
 
             strings
         }
+
+        /// converts the byte strings to rust strings, handling the special case _m (0x00) and _f (0xff) values
+        fn bytes_to_strings(bytes: Vec<u8>) -> String {
+            match bytes.as_slice() {
+                [0] => return "MEASUREMENT(0x00)".into(),
+                // note this isn't valid UTF8 and thus will assert below
+                [255] => return "FIELD(0xff)".into(),
+                _ => {}
+            };
+
+            String::from_utf8(bytes).expect("string value response was not utf8")
```

Comment:
The `_ => {}` case is a bit strange... could this be:

```suggestion
            match bytes.as_slice() {
                [0] => "MEASUREMENT(0x00)".into(),
                // note this isn't valid UTF8 and thus would assert below
                [255] => "FIELD(0xff)".into(),
                _ => String::from_utf8(bytes).expect("string value response was not utf8"),
            }
```

Reply:
Weird -- I swear when I did that I ran into trouble with the borrow checker.... But now that I tried it locally it works just fine. 🤷  this suggestion is much nicer. Thank you